### DISCUSSION TITLE
Idea for Webviz Reservoir Viewer Plugin

### DISCUSSION THREAD

**Discussion by izzy-el** [equinor/webviz-config#583] [General]
Hello, @anders-kiaer! I am Izael and I am writing this discussion with my colleague Vitória, @vikaparra. We are both undergraduate students from Unicamp (University of Campinas). We are working on the Energy Production and Innovation Center (EPIC), under the supervision of Prof. Celmar Guimarães da Silva and with grants provided by Equinor. We are working on a web-based implementation of a software called ReservoirViewer (previously implemented in Java), which uses visualization techniques to represent sets of reservoir models.

The goal of our project is to make available these techniques within Webviz. At first, we were instructed to develop a Python or JavaScript version of ReservoirViewer, so it can be used within Webviz. However, considering the performance of the whole process of generating the visualizations (the Java version usually spends minutes generating each visualization), we thought about implementing the ReservoirViewer in another programming language rather than the ones we cited above. Thus, we considered using Julia Programming Language to implement this work, since it is known for being more performative than Python, JavaScript and Java in most operations. Another option is to connect the current Java-based version of ReservoirViewer to the Webviz.

We discussed both possibilities with Prof. Celmar, and he brought up them in a meeting with the Equinor Brasil team (Rodolfo and William). They instructed us to start this GitHub discussion and talk to you and the other Webviz repository maintainers about these possibilities.

What do you think about using Julia or another programming language rather than Python to implement these visualization resources? Would it be a viable option? Do you consider that inserting a new language would be a problem for Webviz users (considering installation, for example)?

Thank you!

**Comment by anders-kiaer** [comment]
Hi @izzy-el and @Vikaparra. Interesting topic! 🙂 

Could we take a quick video meeting on the topic perhaps, and discuss it a bit? Does one of these times suit you?

- [ ] March 14th 10:00 (São Paulo time zone)
- [ ] March 16th 10:00 (São Paulo time zone)
- [ ] March 18th 10:00 (São Paulo time zone)

Thanks, best regards Anders

**Reply by izzy-el** [reply]
Hey @anders-kiaer. Yeah, I think a meeting would be a good way to discuss this! Out of those 3 days, I am free on March 14th, 10 am.

**Reply by Vikaparra** [reply]
Hey! I'm also free on March 14th, a meeting sounds nice :)

**Reply by anders-kiaer** [reply]
Good. I'll send a teams meeting video link.

**Reply by anders-kiaer** [reply]
@izzy-el Meeting invite sent per e-mail (found your email on the home page), could you forward to @Vikaparra?

**Reply by izzy-el** [reply]
Okay! Thank you @anders-kiaer! I will forward the email to her.

**Comment by magnesj** [comment]
Here is a follow up on the discussions earlier today. ResInsight is used as a post processor for reservoir data, but has also a powerful Python API to some of the features required by your project.

**Contour maps in ResInsight**
Contour maps can be created using one of the predefined  [aggregation algorithms](https://resinsight.org/3d-main-window/contourmaps/#map-projection-properties) This contour map represent the aggregated 2D image of a single realization/model.
These contour maps can be computed by ResInsight and the 2D data for this map can be exported to a text file as shown here
https://api.resinsight.org/en/main/PythonExamples.html#export-contour-maps

Performance is always and issue for these types of computations, and users report that the performance of ResInsight is good.

ResInsight supports Eclipse grid data and the 3D data can be imported into ResInsight using the text format of Eclipse (GRDECL) 
Here is an example of grid geometry in Eclipse GRDECL file format
https://github.com/OPM/opm-tests/blob/5f934211645d08660a480bdbc49a50ad52435899/model2/include/test1.grdecl

These 2D contour maps for each model can then be used as input to the process creating the combined 2D maps for N realizations in the same 2D plot as you showed us in the `ReservoirViewer` tool you have.

### Other resources
https://resinsight.org/3d-main-window/contourmaps/
https://api.resinsight.org/en/main/PythonExamples.html#export-contour-maps

**Reply by izzy-el** [reply]
Alright! Thank you very much! We will take a look at those links and read more about the subject.
